### DISCUSSION TITLE
Import database from faster machine

### DISCUSSION THREAD

**Discussion by arminus** [photoprism/photoprism#1837] [Q&A] [answered]
I'm thinking about running a 24/7 Photoprism instance on a Pi4, for everyday browsing and smaller sized rescans it's probably fast enough. But to index and face recognize ect. my collection of > 30K images it's way too slow.

So I'm thinking about running the initial indexing on a fast box and then load the index database into the Pi4 instance. (probably via a mariadb DB dump/restore). Let's assume the path to the library is identical on both boxes (it's an NFS mounted share from a NAS anyway).

Is that feasible? Or are there any blockers I'm not aware of?

**Comment by lastzero** [comment, marked as answer]
If you know how to do a database and file backup, there are no blockers. It's a standard MariaDB database, so nothing special. The only downside is the work time involved.

**Comment by arminus** [comment]
I can confirm that this works - for anybody searching for this later:

Dump on machine 1: (container names and user/password might vary of course)
```
docker exec -i admin_mariadb_1 mysqldump -u photoprism -pinsecure --routines photoprism > photoprism.sql
```

Load on machine 2:
```
docker exec -i photoprism_mariadb_1 mysql -u photoprism -pinsecure <<EOF
> DROP DATABASE photoprism;
> CREATE DATABASE photoprism CHARACTER SET = 'utf8mb4' COLLATE = 'utf8mb4_unicode_ci';
EOF
docker exec -i photoprism_mariadb_1 mysql -u photoprism -p photoprism < photoprism.sql
```

There's one glitch, though: I don't get any thumbnails for MP4/AVC1 video files on the Pi4 and a message "No video in supported format and MIME type found" when trying to play them. Everything works ok (in the same browser) if I load the same library form the Ubuntu Server/machine1.

I think this might be a side-effect of the DB load, although I'm not entirely sure, maybe this is an unrelated issue on the Pi4?

**Reply by lastzero** [reply]
FFmpeg should work there as well. Does reindexing videos help? Is it a 32-bit Pi / OS?

**Reply by arminus** [reply]
It's a 64bit Pi os. 

In the logs, I just saw this:
time="2021-12-27T18:23:26Z" level=error msg="thumb: file Test/Camera/VID_20200324_151726062.mp4.jpg is missing"
and that's correct because I have a lower case/upper case mix-up in my volume mount which I hadn't notices so far. Need to fix that and make it identical on both boxes and check what'll happen then.

Which brings me to the question if I shouldn't also copy the storage volume from box1 to box2? or at least the sidecar folder?

**Reply by lastzero** [reply]
Yes, unless you only have JPEGs and render all thumbnails dynamically.

**Reply by lastzero** [reply]
Creating thumbs takes a long time, so it makes sense to copy them if your goal is to save time.

**Reply by arminus** [reply]
Ok, one step further: So I rsynced the sidecar folder from box1 to box2. Behavior in the UI is still the same, although the messages in the log now are like
```
time="2021-12-28T10:55:13Z" level=error msg="thumb: thumbnail not cached"
```
VID_20200324_151726062.mp4.jpg is however definitely physically present. What I saw in the file view though for that video are 2 files which are stacked. Clicking Unstack for that file takes a while and then reveals the thumbnail.

So the question is why does it stack the video and thumbnail jpeg in the first place? (Library -> Stacks -> Place/Time) is checked on both boxes but it shouldn't stack a video file and its thumbnail (and .yml file), should it?

(side note: I think the same kind of issues would happen in a recovery from backup case ?)

2nd question: How can I force creation of thumbnails during indexing?

**Reply by arminus** [reply]
Also: Is it enough to replicate the sidecar folder or should I better replicate the entire storage folder?

**Reply by lastzero** [reply]
As mentioned above, you can either recreate thumbnails by running the CLI command as described in our docs, perform a full rescan, or copy the cache folder from the other server (it contains the thumbnails and exiftool json files, which would otherwise also need to be recreated).

**Reply by lastzero** [reply]
The time we spend explaining this would pay for many powerful servers BTW 👯  That's what I mean with work time as "only" downside...

**Reply by lastzero** [reply]
It's really only for advanced users. Our docs contain answers to many of your questions if you read them carefully. For example:

> Cache, session, thumbnail, and sidecar files will be created in the storage folder. Never remove the volume from your docker-compose.yml file so that you don't lose these files after restarting or upgrading the container. We recommend placing the storage folder on a local SSD drive for best performance.

see https://docs.photoprism.app/getting-started/docker-compose/#photoprismstorage

**Reply by lastzero** [reply]
Bottom line: Please copy all files as you otherwise don't save time as you need to recreate them or get stuck with messages informing you that files are missing and you didn't enable dynamic previews at the same time, so they can't be created on demand.

**We always do our best to answer all your questions. In return, we ask you to back us on Patreon or GitHub Sponsors if you're not a sponsor yet. Thank you! 💖**

**Reply by arminus** [reply]
Ok, thanks a lot for your help.
